File: app/src/main/java/com/wlmtest/forecasttestjava/view/forecast/tabs/fivedays/Forecast5DaysFragmentRecicleViewAdapter.java
package com.wlmtest.forecasttestjava.view.forecast.tabs.fivedays;

import android.view.LayoutInflater;
import android.view.ViewGroup;
import androidx.annotation.NonNull;
import androidx.databinding.DataBindingUtil;
import androidx.recyclerview.widget.RecyclerView;
import com.wlmtest.forecasttestjava.R;
import com.wlmtest.forecasttestjava.databinding.Forecast5daysItemBinding;
import com.wlmtest.forecasttestjava.model.pojo.forecastfivedays.CityForecastFiveDaysPojo;
import com.wlmtest.forecasttestjava.viewmodel.MainViewModel;

import java.util.ArrayList;
import java.util.LinkedHashMap;
import java.util.List;


public class Forecast5DaysFragmentRecicleViewAdapter
        extends RecyclerView.Adapter<Forecast5DaysFragmentRecicleViewAdapter.ViewHolder> {

    private MainViewModel mainViewModel;

    public Forecast5DaysFragmentRecicleViewAdapter(MainViewModel mainViewModel) {
        this.mainViewModel = mainViewModel;
    }

    @NonNull
    @Override
    public ViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {

        Forecast5daysItemBinding forecast5daysItemBinding =
                DataBindingUtil.inflate(
                        LayoutInflater.from(parent.getContext()),
                        R.layout.forecast5days_item, parent, false);

        return new ViewHolder(forecast5daysItemBinding);

    }

    @Override
    public void onBindViewHolder(@NonNull ViewHolder holder, int position) {

        LinkedHashMap<String, ArrayList<CityForecastFiveDaysPojo>> forecast5DaysMap =
                this.mainViewModel.getFiveDaysForecastPojoMutableLiveData()
                .getValue()
                .getForecast5DaysMap();

        String dateReference = forecast5DaysMap.keySet()
                .toArray()[position].toString();
        holder.forecast5daysItemBinding
                .setDateReference(dateReference);

        List<CityForecastFiveDaysPojo> cityForecastFiveDaysPojoList = forecast5DaysMap.get(dateReference);
        SmallForecastItemAdapter smallForecastItemAdapter =
                new SmallForecastItemAdapter(cityForecastFiveDaysPojoList);
        holder.forecast5daysItemBinding.setSmallForecastItemAdapter(smallForecastItemAdapter);

    }

    @Override
    public int getItemCount() {
        if (this.mainViewModel.getFiveDaysForecastPojoMutableLiveData() == null ||
                this.mainViewModel.getFiveDaysForecastPojoMutableLiveData().getValue() == null) {
            return 0;
        }
        return this.mainViewModel.getFiveDaysForecastPojoMutableLiveData().getValue().getForecast5DaysMap().size();
    }

    static class ViewHolder extends RecyclerView.ViewHolder {

        public Forecast5daysItemBinding forecast5daysItemBinding;

        public ViewHolder(Forecast5daysItemBinding forecast5daysItemBinding) {

            super(forecast5daysItemBinding.getRoot());
            this.forecast5daysItemBinding = forecast5daysItemBinding;

        }
    }
}

File: app/src/main/java/com/wlmtest/forecasttestjava/view/forecast/tabs/fivedays/Forecast5DaysFragmentRecicleViewAdapter.kt
package com.wlmtest.forecasttestjava.view.forecast.tabs.fivedays

import android.view.LayoutInflater
import android.view.ViewGroup
import androidx.databinding.DataBindingUtil
import androidx.recyclerview.widget.RecyclerView
import com.wlmtest.forecasttestjava.R
import com.wlmtest.forecasttestjava.databinding.Forecast5daysItemBinding
import com.wlmtest.forecasttestjava.viewmodel.MainViewModel


class Forecast5DaysFragmentRecicleViewAdapter(private val mainViewModel: MainViewModel) :
    RecyclerView.Adapter<Forecast5DaysFragmentRecicleViewAdapter.ViewHolder>() {

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ViewHolder {

        val forecast5daysItemBinding = DataBindingUtil.inflate<Forecast5daysItemBinding>(
            LayoutInflater.from(parent.context),
            R.layout.forecast5days_item, parent, false
        )

        return ViewHolder(forecast5daysItemBinding)

    }

    override fun onBindViewHolder(holder: ViewHolder, position: Int) {

        val forecast5DaysMap = this.mainViewModel.fiveDaysForecastPojo
            .value!!
            .forecast5DaysMap

        val dateReference = forecast5DaysMap.keys
            .toTypedArray()[position]
        holder.forecast5daysItemBinding.dateReference = dateReference

        val cityForecastFiveDaysPojoList = forecast5DaysMap[dateReference]
        val smallForecastItemAdapter = SmallForecastItemAdapter(cityForecastFiveDaysPojoList!!)
        holder.forecast5daysItemBinding.smallForecastItemAdapter = smallForecastItemAdapter

    }

    override fun getItemCount(): Int {
        return if (this.mainViewModel.fiveDaysForecastPojo == null || this.mainViewModel.fiveDaysForecastPojo.value == null) {
            0
        } else this.mainViewModel.fiveDaysForecastPojo.value!!.forecast5DaysMap.size
    }

    class ViewHolder(var forecast5daysItemBinding: Forecast5daysItemBinding) :
        RecyclerView.ViewHolder(forecast5daysItemBinding.root)
}
